core/src/main/java/com/datastax/oss/driver/internal/core/channel/ProtocolInitHandler.java

```diff
@@ -140,6 +141,8 @@ protected boolean setConnectSuccess() {
     // This class is a finite-state automaton, that sends a different query depending on the step
     // in the initialization sequence.
     private Step step;
+    private int stepNumber = 0;
```

Comment:
just to double-check, shouldn't this be `volatile`

Reply:
No, because you are inside a class that is only accessed from the event loop, thus it can only be accessed by one thread at a time.